### PR TITLE
Eip1 9474 notifications api spring boot 3 upgrade

### PR DESCRIPTION
**Pull request by Callum-Griffiths**
## Describe your changes
Upgrade to spring boot 3

## Issue ticket number and link

https://dluhcdigital.atlassian.net/browse/EIP1-9474

### CODE REVIEW

File: src/test/resources/application-integration-test.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+spring:
 cloud:
```
Comment (StephenPenney) [inline src/test/resources/application-integration-test.yml:2]: in yaml the lines below the colon should be indented for it to be seen as `spring.cloud.<blah>`

File: src/test/kotlin/uk/gov/dluhc/notificationsapi/messaging/stubs/MessagingSupportLibraryListernerStub.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.slf4j.MDC
+import org.springframework.context.annotation.DependsOn
+import org.springframework.messaging.handler.annotation.Payload
+import org.springframework.stereotype.Component
+import uk.gov.dluhc.logging.config.CORRELATION_ID
+import uk.gov.dluhc.logging.config.MESSAGE_ID
+import uk.gov.dluhc.messagingsupport.MessageListener
+import java.util.Collections
+import java.util.UUID
+@Component
```
Comment (StephenPenney) [inline src/test/kotlin/uk/gov/dluhc/notificationsapi/messaging/stubs/MessagingSupportLibraryListernerStub.kt:13]: it looks like some blank lines have got lost in the copy and paste in various places in this file

File: src/test/kotlin/uk/gov/dluhc/notificationsapi/config/LocalStackContainerConfiguration.kt [hunk 3]
```diff
@@ -69,6 +71,9 @@ class LocalStackContainerConfiguration {
         }
     }
 
+    @Bean
+    fun awsRegionProvider(): AwsRegionProvider = StaticRegionProvider(DEFAULT_REGION)
```
Comment (StephenPenney) [inline src/test/kotlin/uk/gov/dluhc/notificationsapi/config/LocalStackContainerConfiguration.kt:75]: with the idents added to the application-integration-test.yaml file this might not be needed anymore

File: build.gradle.kts [hunk 4]
```diff
@@ -64,12 +64,12 @@ dependencies {
...
+    kapt("org.mapstruct:mapstruct-processor:1.5.5.Final")
 
     // internal libs
-    implementation("uk.gov.dluhc:logging-library:2.3.1")
-    implementation("uk.gov.dluhc:messaging-support-library:1.1.0")
+    implementation("uk.gov.dluhc:logging-library:3.0.3")
+    implementation("uk.gov.dluhc:messaging-support-library:2.1.0")
 
     // api
     implementation("org.springframework.boot:spring-boot-starter-actuator")
```
Comment (StephenPenney) [inline build.gradle.kts:75]: A few lines down (can't comment on that line) there is an openapi-ui dependency, we should remove it and replace with:
```
  implementation("io.swagger.core.v3:swagger-annotations:2.2.7")
  implementation("org.springframework:spring-webmvc")
```

I've now updated the guidance page to include that step as it's a recently reported vulnerability

File: src/test/kotlin/uk/gov/dluhc/notificationsapi/client/ElectoralRegistrationOfficeManagementApiClientTest.kt [hunk 5]
```diff
@@ -70,6 +71,7 @@ internal class ElectoralRegistrationOfficeManagementApiClientTest {
             given(clientResponse.bodyToMono(ElectoralRegistrationOfficeResponse::class.java)).willReturn(
                 Mono.just(eroResponse),
             )
+            given(clientResponse.statusCode()).willReturn(HttpStatus.OK)
```
Comment (StephenPenney) [inline src/test/kotlin/uk/gov/dluhc/notificationsapi/client/ElectoralRegistrationOfficeManagementApiClientTest.kt:74]: do we know why these changes are now required?

**Review by StephenPenney — COMMENTED** [review]
Overall looks good, the docker workflow step at the bottom of the guide needs doing as this repo looks to install Docker 24 due to 25 not working with SB 2.

File: src/test/kotlin/uk/gov/dluhc/notificationsapi/client/ElectoralRegistrationOfficeManagementApiClientTest.kt (hunk 5 above)
Comment (Callum-Griffiths) [reply src/test/kotlin/uk/gov/dluhc/notificationsapi/client/ElectoralRegistrationOfficeManagementApiClientTest.kt:74]: Without it you get error :
   `Cannot invoke "org.springframework.http.HttpStatusCode.value()" because "code" is null`
when `apiClient.getElctoralRegistrationOfficeGssCodes(eroId)` is called
So I think when .bodyToMono is called in `apiClient.getElctoralRegistrationOfficeGssCodes(eroId)` it's no longer mocking a status code.

There's the same changes for similar tests in the Proxy Springboot upgrade https://github.com/communitiesuk/eip-ero-proxy-applications-api/pull/298/files#diff-a704b3d33ec07dc1247fa0d0979b388e500775a01fb7498fcf02cb035e0146f8

**Review by StephenPenney — APPROVED** [review]
This looks great, ready to merge
